### ISSUE TITLE
css-modules for custom properties?

### ISSUE THREAD

**Issue by a-x-** [css-modules/css-modules#303]
Can css-modules add random prefixes for custom properties (aka --css-variables) too?

Now we have possible dangerous global css variables name space.

:global syntax will be useful too.

**Comment by Amareis** [comment]
Yep, that will really useful. Just replace all custom properties definitions and local references until selector is global - it's not hard.

**Comment by soulofmischief** [comment]
How much work is this proposal?

**Comment by blm768** [comment]
#318 seems to be asking for basically the same thing.

**Comment by blm768** [comment]
Two possible hurdles to consider:

* Backward compatibility could be a concern. Adding a configuration flag seems like it'd generally be sufficient to mitigate that, although we wouldn't have scoped custom properties by default in that case.
* Referencing the same custom property across multiple files (which I imagine would be common enough) would probably require some way of "importing" a hashed name from another module (see #342).
